### ISSUE TITLE
nomkl & pyfftw on Windows

### ISSUE THREAD

**Issue by i3v** [adler-j/odlworkshop#17]
`conda install nomkl` and `conda install -c conda-forge pyfftw` currently fail for me on Windows with `PackageNotFoundError`. But at least some stuff from the "part1" seem to work. Are they optional actually?

**Comment by adler-j** [comment]
* `nomkl` is optional on windows.
* `pyfftw` can be replaced by the numpy fft for all exercises, however if you want to do three dimensional stuff it might become somewhat slow. You should be able to install it using `pip install pyfftw` (see [here](https://pypi.python.org/pypi/pyFFTW))

**Comment by kohr-h** [comment]
Actually, `nomkl` fails on Windows, and it seems that the conda package has an issue with a false `nomkl` dependency in it. We need to fix that.

**Comment by i3v** [comment]
Indeed, `pyfftw` seem to be a good choice. It's about 5x faster than numpy (default version from conda) on my machine (for single-axis fft on 512x512x512 array). But, just-in-case: plain `pip install pyfftw` still [may not work](https://github.com/pyFFTW/pyFFTW/issues/168#issue-221495878) on Windows. [Manually downloading](https://github.com/pyFFTW/pyFFTW/issues/168#issuecomment-308049448) correct version do works for me.

**Comment by kohr-h** [comment]
You can also use `conda install -c conda-forge fftw pyfftw`. I don't know if they have the packages in the windows channel, but it's easier to do it manually.

**Comment by i3v** [comment]
Sadly,  `conda install -c conda-forge fftw pyfftw` doesn't work. It [looks](https://anaconda.org/conda-forge/pyfftw) like only osx and linux-64 versions are currently available in "conda-forge" channel. 

[There is](https://anaconda.org/search?q=platform%3Awin-64+pyfftw) one channel that got recent pyfftw version (v0.10.4, they say) for windows - "ccpi". I've decided to test it, and... ran into ["PyFFTW < 0.10.4"](https://github.com/odlgroup/odl/blob/87e04941f4337de190654897691327f8964c0f03/odl/trafos/backends/pyfftw_bindings.py#L26-L30) warning. Somehow, `pyfftw.__version__` gives `0.10.3.dev0+e827cb5`. Thus, installing PyFFTW from that channel doesn't look like a good idea either.

Thus, it looks like a quick-and-dirty workaround with manually downloading `whl` from bintray is the simplest approach for now.

**Comment by kohr-h** [comment]
> There is one channel that got recent pyfftw version (v0.10.4, they say) for windows - "ccpi". I've decided to test it, and... ran into "PyFFTW < 0.10.4" warning. Somehow, pyfftw.__version__ gives 0.10.3.dev0+e827cb5. Thus, installing PyFFTW from that channel doesn't look like a good idea either.

Don't worry about that. The pyfftw devs messed up their versioning, and the 0.10.3 release has the exact same version string as the 0.10.4 release, which makes them indistinguishable at runtime. See also [this issue](https://github.com/odlgroup/odl/issues/1057).

But if you found a variant that works for you, it's all good.
